### PR TITLE
Register additional components for fill process (background removal & vectorizer)

### PR DESCRIPTION
**Pull request by maerch**
Adding additional components that can be added in the dock, inspector bar, navigation bar, canvas bar, and dock.

### CODE REVIEW

File: packages/plugin-utils/src/processing/registerFillProcessingComponents.ts [hunk 1]
```diff
@@ -21,62 +35,140 @@ export default function registerFillProcessingComponents(
...
   const metadata = new FillProcessingMetadata(cesdk.engine, pluginId);
 
-  const canvasMenuComponentId = getCanvasMenuComponentId(pluginId);
   const canvasMenuLabel = getI18nCanvasMenuLabel(pluginId);
+  const canvasMenuComponentIds = getCanvasMenuComponentIds(pluginId);
+  const canvasMenuComponentId = canvasMenuComponentIds[0];
+
+  const dockLabel = getI18nDockLabel(pluginId);
+  const dockComponentIds = getDockComponentIds(pluginId);
+  const dockComponentId = dockComponentIds[0];
```
Comment (mirko314) [inline packages/plugin-utils/src/processing/registerFillProcessingComponents.ts:57]: This "looks" weird. Why would we get a list but only consider the first one?

File: packages/plugin-utils/src/processing/registerFillProcessingComponents.ts [hunk 2]
```diff
@@ -91,10 +183,97 @@ export default function registerFillProcessingComponents(
...
+    },
+    {
+      componentIds: canvasMenuComponentIds,
+      variant: 'plain',
+      label: canvasMenuLabel
+    }
+  ];
+
+  buttonComponents.forEach(({ componentIds, label, variant }) => {
+    const componentId = componentIds[0];
```
Comment (mirko314) [inline packages/plugin-utils/src/processing/registerFillProcessingComponents.ts:214]: Same here, seems weird that we have a list but always only consider the first entry.

**Review by mirko314 — COMMENTED** [review]
LGTM in general - just unsure about the implicit usage of only the first Entry of IDs.
Have not tested it though

File: packages/plugin-utils/src/processing/registerFillProcessingComponents.ts (hunk 1 above)
Comment (maerch) [reply packages/plugin-utils/src/processing/registerFillProcessingComponents.ts:57]: We use the first one for the internal ids. We register the component for both ids.

File: packages/plugin-utils/src/processing/registerFillProcessingComponents.ts (hunk 2 above)
Comment (maerch) [reply packages/plugin-utils/src/processing/registerFillProcessingComponents.ts:214]: In my debugging frenzy, I think I missed rolling back the registering to all ids. Fixed in d56055a

**Comment by hiyeonKim** [comment]
I tested it works, just two things.

- 'bottom' misplaced with 'top' somewhere, which I already requested a review for.
- In `packages/plugin-utils/src/types/UserInterfaceConfiguration.ts`, the type `Location` should be as follows, I think:
```
export type Location =
  | 'inspectorBar'
  | 'navigationBar'
  | 'canvasBarTop'
  | 'canvasBarBottom'
  | 'canvasMenu'
  | 'dock';
```

**Review by hiyeonKim — CHANGES_REQUESTED** [review]
I tested it works, yet,

- In `packages/plugin-utils/src/types/UserInterfaceConfiguration.ts`, the type `Location` should be as follows, I think:
```
export type Location =
  | 'inspectorBar'
  | 'navigationBar'
  | 'canvasBarTop'
  | 'canvasBarBottom'
  | 'canvasMenu'
  | 'dock';
```

**Comment by maerch** [comment]
> I tested it works, yet,
> 
> * In `packages/plugin-utils/src/types/UserInterfaceConfiguration.ts`, the type `Location` should be as follows, I think:
> 
> ```
> export type Location =
>   | 'inspectorBar'
>   | 'navigationBar'
>   | 'canvasBarTop'
>   | 'canvasBarBottom'
>   | 'canvasMenu'
>   | 'dock';
> ```

Good catch. Changed in edf6a9b

**Review by hiyeonKim — APPROVED** [review]

**Pull request merged**